src/Features/WetnessEffects.h

```diff
@@ -31,27 +31,54 @@ struct WetnessEffects : Feature
 		float MinRainWetness = 0.65f;
 		float SkinWetness = 0.825f;
 		float WeatherTransitionSpeed = 3.0f;
+
+		// Raindrop fx settings
+		uint EnableRaindropFx = true;
+		uint EnableSplashes = true;
+		uint EnableRipples = true;
+		uint EnableChaoticRipples = true;
+		float RaindropFxRange = 1000.f;
+		float RaindropGridSize = 3.f;
+		float RaindropInterval = 1.f;
+		float RaindropChance = .3f;
+		float SplashesStrength = 1.2f;
+		float SplashesMinRadius = .3f;
+		float SplashesMaxRadius = .7f;
+		float RippleStrength = .5f;
+		float RippleRadius = 1.f;
+		float RippleBreadth = .5f;
+		float RippleLifetime = .1f;
+		float ChaoticRippleStrength = .1f;
+		float ChaoticRippleScale = 1.f;
+		float ChaoticRippleSpeed = 20.f;
 	};
 
 	struct alignas(16) PerPass
 	{
+		float Time;
+		uint Raining;
 		float Wetness;
 		float PuddleWetness;
 		DirectX::XMFLOAT3X4 DirectionalAmbientWS;
+		RE::DirectX::XMFLOAT4X4 PrecipProj;
```

Comment:
Do we need the RE:: prefix?

Reply:
Its value is copied from a CommonLib which requires this namespace, otherwise the compiler doesn't regard them as the same type. See https://github.com/doodlum/skyrim-community-shaders/blob/da2ec4a526ba0b048d1c739060f25615a25b8e5b/src/Features/WetnessEffects.cpp#L319

Reply:
It's just weird that directly above it we don't use RE:.

Reply:
imo it's weird that clib partially copies a implementation of a class from a well-available external library and put it under its namespace.
https://github.com/alandtse/CommonLibVR/blob/0b92b1aad79c22b9a0620c4bdd244e9f4ad94fea/include/SKSE/Impl/WinAPI.h#L798

Reply:
i think it's fine. but you could add a way to clib for it to link with directx surely